README.md

```diff
@@ -1,50 +1,96 @@
 
-## Usage
+- Each month, the Gelato bot proposes a new root. For 3 days, I have the opportunity to run checks on this root. After these 3 days, anyone can accept this value.
+- The DAO must transfer the correct amount of tokens to the URD to allow all claimants to claim their rewards. If the DAO does not provide enough funds, the claim function will fail with the message "not enough funds in URD."
 
-Merkle trees should be generated with [Openzeppelin library](https://github.com/OpenZeppelin/merkle-tree).
-It will ensure that trees will be secure for on-chain verification.
+## Attaching an IPFS Hash
 
-## Installation
+- Using a Merkle tree delegates all root computation off-chain, leaving no information about the tree on-chain (except for the root). This makes it challenging for an integrator (or a claimer) to understand the Merkle tree or to know which proof to use. To address this, each root can be linked to an IPFS hash, which can link to any data. We recommend that all users follow the same format for the IPFS hash to facilitate integration. The suggested format is as follows:
 
-Download foundry:
-```bash
-curl -L https://foundry.paradigm.xyz | bash
+```json
+{
+  "id": "A string id of the merkle tree, can be random",
+  "metadata": {
+  "info": "a key value mapping allowing you to add information"
+  },
+  "root": "The merkle root of the tree",
+  "tree": [
+       {
+              "account": "The address of the claimer",
+              "reward": "The address of the reward token",
+              "claimable": "The claimable amount as a big number",
+              "proof": ["0x1...", "0x2...", "...", "0xN..."]
+      }
+  ]
+}
 ```
 
-Install it:
-```bash
-foundryup
-```
+- We recommend not including spaces or new lines when uploading to IPFS, to ensure a consistent method of uploading your JSON file.
 
-Install dependencies:
-```bash
-git submodule update --init --recursive
-```
+## Owner Specifications
 
-Now you can run tests, using forge:
-```bash
-forge test
-```
+- The URD is an owner-managed contract, meaning the owner has full control over its distribution. Specifically, the owner can bypass all time-locked functions, modify the time-lock, add or remove updaters, and revoke the pending value at their discretion.
+- If the owner is set to a zero address, the contract becomes ownerless. In this scenario, only updaters can submit a root. Furthermore, the time-lock value becomes unchangeable, and pending values cannot be removed.
+- If there are neither owner nor updaters, the URD becomes immutable. This is useful for creating a one-time distribution that must remain unchanged. This can be accomplished at the contract's creation by providing only a Merkle root and an optional IPFS hash. After this, the contract's sole functionality is to claim rewards.
```

Comment:
The URD is already immutable, when funds are deposited and the root has been submitted, it's trustless to me

Reply:
Not really imo, owner can still rug users. So how define a contract as trustless if someone can rug you?

Reply:
how?

Reply:
Oh ok got it, changing it